game_mechanics.py

```diff
@@ -79,23 +81,119 @@ def _make_move(board: np.ndarray, move: Tuple[int, int], current_player: int) ->
 def is_legal_move(board: np.ndarray, move: Tuple[int, int], current_player: int) -> bool:
     board_dim = board.shape[0]
     if is_valid_coord(board_dim, move[0], move[1]) and board[move] == 0:
-        for direction in MOVE_DIRS:
-            if has_tile_to_flip(board, move, direction, current_player):
-                return True
+        if has_tile_to_flip(board, move, current_player):
+            return True
     return False
 
 
+def idx_surrounding(arr: np.ndarray, idx: Tuple[int, int]) -> Tuple[np.ndarray, np.ndarray]:
+    """Returns he indexes of the elements within a radius of 1 of the original element (arr must be
+    2d and square currently)"""
+    assert arr.shape[0] == arr.shape[1]
+    dim = arr.shape[0]
+    x, y = idx
+    for num in (x, y):
+        assert 0 <= num < dim
```

Comment:
I can't stop myself :stuck_out_tongue_winking_eye: 

I'm off the chain! :chains: 
```suggestion
    assert all(0 <= num < dim for num in (x, y))
```